src/Mutation.php

```diff
@@ -57,14 +57,26 @@
      */
     private $hash;
 
+    /**
+     * @var Node
+     */
+    private $mutatedNode;
+
+    /**
+     * @var int
+     */
+    private $mutationByMutatorIndex;
+
     public function __construct(
         string $originalFilePath,
         array $originalFileAst,
         Mutator $mutator,
         array $attributes,
         string $mutatedNodeClass,
         bool $isOnFunctionSignature,
-        bool $isCoveredByTest
+        bool $isCoveredByTest,
+        Node $mutatedNode,
+        int $mutationByMutatorIndex
```

Comment:
This class now needs 9 parameters to instantiate. Its probably outside the scope of this PR, but we can probably refactor this to reduce that number, and keep it maintanable.

Reply:
From the first look, I don't see how we can reduce the number of parameters (I'm not speaking about grouping them to some value objects which is the same basically).

Almost all of the public methods are just pure getters and. If you have any particular ideas, please advice.